### ISSUE TITLE
Selective-AC remat rejects SDPA even with dropout_p=0 due to nondeterministic_seeded tag

### ISSUE THREAD

**Issue by wconstab** [pytorch/pytorch#187985]
`torch._functorch.partitioners.has_recomputable_rng_ops` flags any recomputable node whose `node.target.tags` contains `torch.Tag.nondeterministic_seeded`. 

All four SDPA ops — `scaled_dot_product_attention`,
  `_scaled_dot_product_flash_attention`, `_scaled_dot_product_efficient_attention`, and `_scaled_dot_product_fused_attention_overrideable` — carry that tag, so a dropout_p=0 SDPA is treated as RNG.

As a result, the forward-loss-backward remat pass (`torch/_functorch/_activation_checkpointing/remat_using_tags_for_fwd_loss_bwd_graph_pass.py`) raises **RuntimeError("...does not support RNG ops in recompute regions... use joint graph mode")**, even though a dropout-free SDPA draws no randomness and is bitwise-deterministic to recompute.

The min-cut joint partitioner avoids this via functionalize_rng_ops (it threads explicit Philox state so a recomputed seeded op replays from the saved seed). The forward-loss-backward remat path has no such escape, so it conservatively errors.

Repro:
```
import torch
import torch.nn.functional as F
from torch.fx.experimental.proxy_tensor import make_fx
from torch._functorch.partitioners import has_recomputable_rng_ops

def f(q, k, v):
    return F.scaled_dot_product_attention(q, k, v, dropout_p=0.0)

q = k = v = torch.randn(2, 4, 512, 64, requires_grad=True)
gm = make_fx(f)(q, k, v)

# Mark the SDPA node as must-recompute (what selective AC does), then check:
for node in gm.graph.nodes:
    if node.op == "call_function" and "scaled_dot_product_attention" in str(node.target):
        node.meta["recompute"] = torch.utils.checkpoint.CheckpointPolicy.PREFER_RECOMPUTE

print(has_recomputable_rng_ops(gm))   # -> True, despite dropout_p=0.0
```

Edit: a note on motivation:
I'm running into this because I'm using full-graph capture (min_viable_fx) on torchtitan qwen models using SAC.  I think that explains why a lot of people haven't run into this (torch.compile path via min-cut partitioner handles this already, as does eager+AC).  I'm not sure if graphtrainer folks ran into this (cc @SherlockNoMad @xmfan) but @aditvenk did mention autoparallel may have worked around this using graphsafe_run_with_rng_state - i'll check whether that helps in my case. 

cc @soulitzer @chauhang @penguinwu @drisspg @liangel-02 @howardzhang-cv
